### PR TITLE
Create Telegram bot

### PR DESCRIPTION
**Pull request by bartekpacia**
> I have many important school things to do, so let's spend several hours doing something completely unrelated

Let's support Telegram!

The fact that the project & code structure is so similar to [LangPolizei](https://github.com/bartekpacia/telegram-lang-enforcer) should make it fast and easy to do. Thanks for embracing it.

### CODE REVIEW

**Comment by bartekpacia** [comment]
eslint version is valid now, idk why he's so angry

**Comment by diogoscf** [comment]
> eslint version is valid now, idk why he's so angry

It's because the bot doesn't support the latest version of eslint sigh, I'll replace it when I have time, but ignore it in the meantime

File: src/telegram/bot_telegram.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+   */
+  start(): void {
+    // The essence of this bot, scan all messages
+    this.on("message", async (msg) => {
+      if (msg.text === undefined) {
+        console.log("Message doesn't contain text, returned. (msg.text === undefined)");
+        return;
+      }
+
+      if (msg.chat.type === "private") {
```
Comment (diogoscf) [inline src/telegram/bot_telegram.ts:31]: tbh I think github lines should work in DMs as well

Comment (bartekpacia) [reply src/telegram/bot_telegram.ts]: thanks - indeed, it works well in private message after I allowed them

**Comment by bartekpacia** [comment]
@diogoscf review plz

File: .eslintrc.json [hunk 2]
```diff
@@ -46,21 +59,33 @@
     ],
     "import/newline-after-import": "off",
     "@typescript-eslint/no-use-before-define": "off",
-    "@typescript-eslint/camelcase": ["error", { "properties": "never" }],
-    "@typescript-eslint/interface-name-prefix": ["error", { "prefixWithI": "always" }],
+    "@typescript-eslint/camelcase": "off",
+    "@typescript-eslint/interface-name-prefix": "off",
```
Comment (diogoscf) [inline .eslintrc.json:63]: Why'd you turn these off

File: src/telegram/bot_telegram.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  }
+
+  /**
+   * Start listening to Telegram webhook.
+   */
+  start(): void {
+    // The essence of this bot, scan all messages
+    this.on("message", async (msg) => {
+      if (!msg.text) {
+        console.log("Message doesn't contain text, returned. (msg.text === undefined)");
```
Comment (diogoscf) [inline src/telegram/bot_telegram.ts:27]: What's the point of this console.log

File: src/telegram/bot_telegram.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    }
+
+    const messages = msgList.map(
+      (ld) => `\`\`\`${ld.toDisplay.search(/\S/) !== -1 ? ld.extension : " "}\n${ld.toDisplay}\n\`\`\``
+    );
+
+    const botMsg = messages.join("\n") || null;
+
+    if (botMsg && botMsg.length >= 2000) {
+      return ["Sorry but there is a 2000 character limit on Discord, so we were unable to display the desired snippet"];
```
Comment (diogoscf) [inline src/telegram/bot_telegram.ts:84]: Not really discord :P

**Review by diogoscf — COMMENTED** [review]
Just a couple things, otherwise lgtm

File: src/telegram/bot_telegram.ts (hunk 3 above)
Comment (bartekpacia) [reply src/telegram/bot_telegram.ts:27]: "debugging purposes", but yeah, it doesn't make much sense so I removed it

File: .eslintrc.json (hunk 2 above)
Comment (bartekpacia) [reply .eslintrc.json:63]: I was getting warnings that made no sense:
Screenshot 2020-12-12 at 14 18 08


I googled and it's [this issue](https://github.com/typescript-eslint/typescript-eslint/issues/2077). The fix I found was to just turn off those rules (they aren't of much use anyway)

File: src/telegram/bot_telegram.ts (hunk 4 above)
Comment (bartekpacia) [reply src/telegram/bot_telegram.ts]: max length changed 4096 (and name to Telegram, ofc) 

[src](https://www.google.com/search?q=telegram+message+character+limit)

**Review by diogoscf — APPROVED** [review]
LGTM

**Comment by diogoscf** [comment]
Telegram Bot won't be up for a while though, I have exams and then vacations

**Pull request merged**